tests/pytests/test_empty.py

```diff
@@ -0,0 +1,641 @@
+        # returned
+        cmd = f'FT.SEARCH {idx} -isempty(@t)'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # Search for a negation of a non-empty value, make sure the document is
+        # returned
+        cmd = f'FT.SEARCH {idx} -@t:{{foo}}'.split(' ')
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Union --------------------------------
+        # Union of empty and non-empty values
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) | @t:{foo}']
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ---------------------------- Intersection ----------------------------
+        # Intersection of empty and non-empty values
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) @t:{foo}']
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Prefix -------------------------------
+        # We shouldn't get the document when searching for a prefix of "__empty"
+        cmd = f'FT.SEARCH {idx} @t:{{*pty}}'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Suffix -------------------------------
+        # We shouldn't get the document when searching for a suffix of "__empty"
+        cmd = f'FT.SEARCH {idx} @t:{{__em*}}'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # Add a document that will be found by the suffix search
+        conn.execute_command('HSET', 'h2', 't', 'empty')
+        cmd = f'FT.SEARCH {idx} @t:{{*pty}}'.split(' ')
+        expected = [1, 'h2', ['t', 'empty']]
+        cmd_assert(env, cmd, expected)
+        conn.execute_command('DEL', 'h2')
+
+        # -------------------- Combination with other fields -------------------
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['hello | isempty(@t)']
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, [1, 'h1', ['t', '']])
+
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['hello isempty(@t)']
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, EMPTY_RESULT)
+
+        # Non-empty intersection with another field
+        conn.execute_command('HSET', 'h1', 'text', 'hello')
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['hello isempty(@t)']
+        expected = [1, 'h1', ['t', '', 'text', 'hello']]
+        cmd_assert(env, cmd, expected)
+
+        # Non-empty union with another field
+        conn.execute_command('HSET', 'h2', 'text', 'love you', 't', 'movie')
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['love | isempty(@t)', 'SORTBY', 'text', 'ASC']
+        expected = [
+            2, 'h1', ['text', 'hello', 't', ''], 'h2', ['text', 'love you', 't', 'movie']
+        ]
+        cmd_assert(env, cmd, expected)
+
+        if dialect == 2:
```

Comment:
Need to be removed after fixing [MOD-6967](https://redislabs.atlassian.net/browse/MOD-6967?atlOrigin=eyJpIjoiOTQ2NjdkZjRlYThhNGUzZTk5YTRiNTFlNGM0Y2FiNzIiLCJwIjoiamlyYS1zbGFjay1pbnQifQ) (`FT.AGGREGATE` fails if apply and sortby with dialect > 2)